datafusion/core/tests/memory_limit.rs

```diff
@@ -581,7 +581,9 @@ fn make_dict_batches() -> Vec<RecordBatch> {
         // ...
         // 0000000002
 
-        let values: Vec<_> = (i..i + batch_size).map(|x| format!("{x:010}")).collect();
+        let values: Vec<_> = (i..i + batch_size)
+            .map(|x| format!("{:010}", x / 16))
```

Comment:
This change was necessary to make it so the dictionaries contain some repeated values, which avoids the RowConverter overheads coming to dominate the memory usage to the point where testing the spill reservation becomes a moot point